tests/tests.rs

```diff
@@ -112,49 +111,62 @@ fn should_verify_schnorr_bip340_secp256k1_signature() {
 #[test]
 fn should_verify_schnorr_bip341_secp256k1_signature() {
     let canister = CanisterSetup::default();
+    let message_sizes = [0, 1, 32, 100, 10_000, 2_000_000];
+
+    for message_size in message_sizes {
+        let derivation_path = vec!["test-derivation-path".as_bytes().to_vec()];
+        let key_id = SchnorrKeyId {
+            algorithm: SchnorrAlgorithm::Bip340Secp256k1,
+            name: "insecure_test_key_1".to_string(),
+        };
+        let message = vec![0xAB; message_size];
+
+        let public_key_raw = canister
+            .schnorr_public_key(SchnorrPublicKeyArgs {
+                canister_id: None,
+                derivation_path: derivation_path.clone(),
+                key_id: key_id.clone(),
+            })
+            .public_key;
+
+        let merkle_root_hash_bytes = vec![0xCD; 32];
+
+        let aux = SignWithSchnorrAux::Bip341(SignWithBip341Aux {
+            merkle_root_hash: ByteBuf::from(merkle_root_hash_bytes.clone()),
+        });
+
+        let signature_raw = canister
+            .sign_with_schnorr(SignWithSchnorrArgs {
+                message: message.clone(),
+                derivation_path,
+                key_id,
+                aux: Some(aux),
+            })
+            .signature;
+
+        let pk = XOnlyPublicKey::from_slice(&public_key_raw[1..]).unwrap();
+
+        let secp256k1_engine = Secp256k1::new();
+        let merkle_root = Some(
+            bitcoin::hashes::Hash::from_slice(&merkle_root_hash_bytes)
+                .expect("failed to create TapBranchHash"),
+        );
 
-    let derivation_path = vec!["test-derivation-path".as_bytes().to_vec()];
-    let key_id = SchnorrKeyId {
-        algorithm: SchnorrAlgorithm::Bip340Secp256k1,
-        name: "insecure_test_key_1".to_string(),
-    };
-    let message = vec![0xAB; 32];
-
-    let public_key_raw = canister
-        .schnorr_public_key(SchnorrPublicKeyArgs {
-            canister_id: None,
-            derivation_path: derivation_path.clone(),
-            key_id: key_id.clone(),
-        })
-        .public_key;
-
-    let merkle_root_hash_bytes = vec![0xCD; 32];
-
-    let aux = SignWithSchnorrAux::Bip341(SignWithBip341Aux {
-        merkle_root_hash: ByteBuf::from(merkle_root_hash_bytes.clone()),
-    });
-
-    let signature_raw = canister
-        .sign_with_schnorr(SignWithSchnorrArgs {
-            message: message.clone(),
-            derivation_path,
-            key_id,
-            aux: Some(aux),
-        })
-        .signature;
-
-    let pk = XOnlyPublicKey::from_slice(&public_key_raw[1..]).unwrap();
+        // The `secp256k1` crate allows for verifitcation of arbitrary-sized messages,
+        // but the `bitcoin` uses an older version. In the following, convert
+        // `XOnlyPublicKey` from `bitcoin`'s `secp256k1` to `secp256k1`'s `XOnlyPublicKey`
```

Comment:
How about: "Because of this, we need to convert from `bitcoin::XOnlyPublicKey` to `secp256k1::XOnlyPublicKey`."